spec/features/cases/offender_sar_complaint/case_creating_spec.rb

```diff
@@ -4,19 +4,18 @@
   given(:manager)         { find_or_create :branston_user }
   given(:managing_team)   { create :managing_team, managers: [manager] }
   given(:offender_sar) { create :offender_sar_case, :third_party, :closed}
+  given(:complaint_type) { 'standard_complaint' }
+  given(:complaint_subtype) { Case::SAR::OffenderComplaint.complaint_subtypes.keys().sample }
+  given(:complaint_priority) { Case::SAR::OffenderComplaint.priorities.keys().sample }
+  given(:default_params) { { complaint_type: complaint_type, complaint_subtype: complaint_subtype, priority: complaint_priority } }
+  given(:target_case) { build(:offender_sar_complaint, default_params).decorate }
 
```

Comment:
@Polsonby,  if we setup the basic types like complaint_type at the beginning of test spec, then it will lose the intention of practising the possible combination of those types cross different tests within this test spec, the coverage of possible scenarios will be significantly less.   I would still prefer to generate the choices dynamically among this test spec unless there is reason for not being able to do so.

Reply:
I think there were a couple of reasons for this - one was that when I added in the ability to set and check for the contacts, the tests started failing randomly because there's a dependency between complaint types and expectations of what the contact details part of the page should contain. I'm struggling to remember properly now after the gap, but I think it was something along those lines. Was conscious that this reduced the test variations but  I'm not sure how I feel about randomised tests and expectations; I'd much rather tests were deterministic and always create the same expectations and test setup on every run. Not sure how much value the random coverage actually provides in practice?

Reply:
However, I'm sure there's a better way to structure the tests than what I've done, maybe it's something to iterate on